From e895d93f4c3b758fe8ed5c8e656b2c80b1400d3c Mon Sep 17 00:00:00 2001
From: Dan Callaghan <dcallagh@redhat.com>
Date: Fri, 3 Aug 2012 12:21:40 +1000
Subject: [PATCH 1/8] gifwriter bgcolor narrowing
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gif/gifwriter.cpp: In function ‘bool exportGif(const QImage&, const QString&)’:
gif/gifwriter.cpp:109:47: warning: narrowing conversion of ‘bgcolor’ from ‘int’ to ‘char’ inside { } is ill-formed in C++11 [-Wnarrowing]

diff --git a/src/gif/gifwriter.cpp b/src/gif/gifwriter.cpp
index c0a010c..27c66c7 100644
--- a/src/gif/gifwriter.cpp
+++ b/src/gif/gifwriter.cpp
@@ -106,7 +106,7 @@ bool exportGif(const QImage &img, const QString &fileName)
     gif->ImageCount = 1;
     EGifPutScreenDesc(gif, image.width(), image.height(), 256, 0, &cmap);
     if (bgcolor >= 0) {
-        char extension[] = { 1, 0, 0, bgcolor };
+        unsigned char extension[] = { 1, 0, 0, (unsigned char) bgcolor };
         EGifPutExtension(gif, GRAPHICS_EXT_FUNC_CODE, 4, extension);
     }
     EGifPutImageDesc(gif, 0, 0, image.width(), image.height(), 0, &cmap);
-- 
1.7.11.7

